src/runtime_src/core/edge/user/hwctx_object.cpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+#include "hwctx_object.h"
+#include "graph_object.h"
+#include "shim.h"
+
+namespace zynqaie {
+  // Shim handle for hardware context Even as hw_emu does not
+  // support hardware context, it still must implement a shim
+  // hardware context handle representing the default slot
+  //
+  //
+  hwctx_object::hwctx_object(ZYNQ::shim* shim, slot_id slotidx, xrt::uuid uuid, xrt::hw_context::access_mode mode)
+	  : m_shim(shim)
+	    , m_uuid(std::move(uuid))
+	    , m_slotidx(slotidx)
+	    , m_mode(mode)
+  {}
```

Comment:
Nice, this is how I prefer to initialized member variables in initializer list one line per variable.
Nit, align `,` with `:`

Reply:
done